django/contrib/admin/templates/admin/change_form.html

```diff
@@ -23,6 +23,33 @@
 {% endblock %}
 {% endif %}
 
+{% block pretitle %}
+  {% if errors %}
+    <p class="errornote">
+      {% blocktranslate count counter=errors|length %}Please correct the error below.{% plural %}Please correct the errors below.{% endblocktranslate %}
+    </p>
+
+    {{ adminform.form.non_field_errors }}
+
+    <div class="errornote">
+      <p>The following fields have errors</p>
```

Comment:
Looking at: https://design-system.w3.org/styles/form-errors.html#error-summary-considerations
I think we want something closer to this:
```suggestion
    <div class="errornote" role="alert" aria-labelledby="error-summary-title" tabindex="-1">
      <h2 id="error-summary-title">
        {% blocktranslate %}There is a problem{% endblocktranslate %}
      </h2>

      {{ adminform.form.non_field_errors }}

      <p>The following fields have errors:</p>
```
Which would look like:
![Screenshot from 2024-09-03 16-19-49](https://github.com/user-attachments/assets/b5efe193-64c3-4b24-ba18-7f1336d1ab9a)
(the header might want a bit of styling around the margin etc)

You're right that changing: "Please correct the error below" would cause some test failures. It's interesting that the W3C is saying it "must" say "There is a problem" (don't feel like it's true, might just need a consistent title)

On the `tabindex="-1"`, it mentions that it has JavaScript to handle this in the example but I can't "spot" the JavaScript in question

We should probably involve the @django/accessibility team heavily around getting the design of this right before worrying too much about updating the other templates/tests

Reply:
Yes. I think adding `tabindex=-1` is nice, but there would need to be some javascript, which on form submit, if there is error detected, will focus this particular element. Negative tabindex don't put the element in the general order, which is nice and recommended, but allows to programmatically focus.

Reply:
Also, @sarahboyce I agree with your suggestion. Honestly, moving above the title is not a strict requirement, as long as it's the first focused element, but I think the w3c recommendation is based on the fact that having it at the top of the page makes it the first thing to navigate to, even if not focused correctly, so definitely a good to have.